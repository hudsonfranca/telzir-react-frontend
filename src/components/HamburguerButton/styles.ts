import styled from 'styled-components';

export interface Props {
  open: boolean;
}
export const Container = styled.div<Props>`
  width: 50px;
  height: 50px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  cursor: pointer;

  div {
    width: 100%;
    height: 3px;
    background-color: var(--white);
    transition: all 0.3s linear;
    transform-origin: 1px;
    border-radius: 10px;

    :first-child {
      transform: ${({ open }) => (open ? 'rotate(45deg)' : 'rotate(0)')};
    }

    :nth-child(2) {
      opacity: ${({ open }) => (open ? '0' : '1')};
      transform: ${({ open }) => (open ? 'translateX(20px)' : 'translateX(0)')};
    }

    :nth-child(3) {
      transform: ${({ open }) => (open ? 'rotate(-45deg)' : 'rotate(0)')};
    }
  }
`;
