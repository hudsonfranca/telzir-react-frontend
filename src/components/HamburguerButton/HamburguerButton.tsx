import React from 'react';

import { Container } from './styles';

export interface Props {
  open: boolean;
  onClick: () => any;
}

const HamburguerButton: React.FC<Props> = ({ open, onClick }) => {
  return (
    <Container
      data-test="hamburguerButtonContainer"
      open={open}
      onClick={onClick}
    >
      <div data-test="hamburguerLine" />
      <div data-test="hamburguerLine" />
      <div data-test="hamburguerLine" />
    </Container>
  );
};

export default HamburguerButton;
