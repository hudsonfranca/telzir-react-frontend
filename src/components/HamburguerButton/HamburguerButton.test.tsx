import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import HamburguerButton from './HamburguerButton';

const mockedHandleClickHbButton = jest.fn();

function createTestProps() {
  return {
    open: false,
    onClick: mockedHandleClickHbButton,
  };
}

let wraper: ShallowWrapper;

beforeEach(() => {
  const props = createTestProps();
  wraper = shallow(<HamburguerButton {...props} />);
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('<HamburguerButton/> redering', () => {
  it('Should render without errors', () => {
    const props = createTestProps();
    const wrapper = renderer.create(<HamburguerButton {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render one container div', () => {
    expect(wraper.find(`[data-test='hamburguerButtonContainer']`).length).toBe(
      1
    );
  });

  it('Should render three hamburguerLine div', () => {
    expect(wraper.find(`[data-test='hamburguerLine']`).length).toBe(3);
  });
});

describe('<HamburguerButton/> interactions', () => {
  it('Should call the oneClick function when the hamburguer button is clicked', () => {
    const hbButton = wraper.find(`[data-test='hamburguerButtonContainer']`);
    hbButton.simulate('click');
    expect(mockedHandleClickHbButton).toHaveBeenCalledTimes(1);
  });
});
