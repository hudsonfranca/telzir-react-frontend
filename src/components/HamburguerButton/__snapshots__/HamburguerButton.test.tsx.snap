// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburguerButton/> redering Should render without errors 1`] = `
.c0 {
  width: 50px;
  height: 50px;
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 div {
  width: 100%;
  height: 3px;
  background-color: var(--white);
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
  border-radius: 10px;
}

.c0 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c0 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c0 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<div
  className="c0"
  data-test="hamburguerButtonContainer"
  onClick={[MockFunction]}
  open={false}
>
  <div
    data-test="hamburguerLine"
  />
  <div
    data-test="hamburguerLine"
  />
  <div
    data-test="hamburguerLine"
  />
</div>
`;
